feat(Mint): add approve ntf payment smart contract

yehia67 opened the pull request.

Commit:
feat(Mint): add approve ntf payment smart contract

Commit:
remove unused hooks

EmanHerawy reviewed (changes requested):
As we have discussed in the call, we need to go to the next step when the approve txn is confirmed

ahmedgalalmohamed2016 reviewed (commented):
in if condition to check nft and confirm that it is not null. we set it null to pass demo but it must be redirect it to the previous step if we found any issues in data

Commit:
fix:
- make user wait till transactionn submitted on MetaMask
-  redirect user regarding transaction status

Commit:
fix: remove unused code
